mm2src/coins/lp_coins.rs

```diff
@@ -2296,9 +2296,17 @@ pub enum ValidatorsInfoDetails {
 }
 
 #[derive(Serialize, Deserialize)]
-pub struct SignatureRequest {
-    coin: String,
-    message: String,
+#[serde(tag = "derivation_method", rename_all = "snake_case")]
+pub enum SignatureRequest {
+    Iguana {
+        coin: String,
+        message: String,
+    },
+    HdWallet {
+        coin: String,
+        message: String,
+        derivation_path: RpcDerivationPath,
+    },
 }
```

Comment:
does it make sense to tell the caller to state whether they want the signing with iguana or HD wallet given that KDF was initialized exclusively with either iguana or HD wallet?

i mean, if i initialized KDF with `enable_hd: true`, signing requests should implicitly be HD based (as signing with iguana should error anyway and vice versa), or let's say, signing should depend on the activated coin (assuming we can have one coin activated in HD mode while another activated in another mode (e.g. pubkey-only)).

what im saying is, the `derivation_method` field shouldn't exist.

Reply:
valid point 👍🏾